client/my-sites/email/email-management/home/utils.js

```diff
@@ -123,6 +125,21 @@ export function resolveEmailPlanStatus( domain, emailAccount, isLoadingEmails )
 			return errorStatus;
 		}
 
+		// When users have registered a domain with us, we let them purchase Google Workspace
+		// before the domain provisioning has finished. However, the user won't see any mailboxes
+		// in the email management until the domain provisioning has finished. To avoid confusion,
+		// we display a warning under these conditions.
+		if (
+			isRecentlyRegistered( domain, 45 ) &&
```

Comment:
I tested this pull request but for some reason this particular warning wouldn't be displayed while the right text would appear in the `Mailboxes` section :

<img width="947" alt="02" src="https://user-images.githubusercontent.com/594356/176204696-ae822791-c35f-448d-97c4-9f79889c426d.png">

It took me a few seconds to find the culprit as we are using the same code in both places (well, almost the same 😀):

```suggestion
			isRecentlyRegistered( domain.registrationDate, 45 ) &&
```

Reply:
🤦 Thanks for the catch!

I made that mistake previously in some other place as well. Added type annotations to that file to prevent that mistake from happening again.